src/systemctl/systemctl-edit.c

```diff
@@ -334,33 +359,50 @@ static int run_editor(char **paths) {
                 if (!editor)
                         editor = getenv("VISUAL");
 
-                if (!isempty(editor)) {
+                if (isempty(editor))
+                        argc = 1;
+                else {
                         editor_args = strv_split(editor, WHITESPACE);
                         if (!editor_args) {
                                 (void) log_oom();
                                 _exit(EXIT_FAILURE);
                         }
                         n_editor_args = strv_length(editor_args);
-                        argc += n_editor_args - 1;
+                        argc = n_editor_args;
                 }
 
-                args = newa(const char*, argc + 1);
+                for (const EditFile *f = files; f->path; f++)
+                        argc += 2;
+
+                args = newa(char*, argc + 1);
 
                 if (n_editor_args > 0) {
                         args[0] = editor_args[0];
                         for (; i < n_editor_args; i++)
                                 args[i] = editor_args[i];
                 }
 
-                STRV_FOREACH_PAIR(original_path, tmp_path, paths)
-                        args[i++] = *tmp_path;
+                if (files[0].path && files[0].line > 1 && !files[1].path) {
+                        /* If editing a single file only, use the +LINE syntax to put cursor on the right line */
```

Comment:
Won't this cause issues with editors that don't know the `+X` syntax? From top of my head I can think of `ed` or `cat`, which is not a very "real world" scenario, but maybe other editors will choke up as well?

```
$ EDITOR=cat sudo -E build/systemctl edit --runtime sshd.service
cat: +4: No such file or directory
### Editing /run/systemd/system/sshd.service.d/override.conf
### Anything between here and the comment below will become the new contents of the file



### Lines below this comment will be discarded

### /usr/lib/systemd/system/sshd.service
# [Unit]
# Description=OpenSSH server daemon
# Documentation=man:sshd(8) man:sshd_config(5)
# After=network.target sshd-keygen.target
# Wants=sshd-keygen.target
# 
# [Service]
# Type=notify
# EnvironmentFile=-/etc/sysconfig/sshd
# ExecStart=/usr/sbin/sshd -D $OPTIONS
# ExecReload=/bin/kill -HUP $MAINPID
# KillMode=process
# Restart=on-failure
# RestartSec=42s
# 
# [Install]
# WantedBy=multi-user.target(editor) failed with exit status 1.
```

or 

```
# Old systemctl
$ echo -ne '2a\nhello\n.\nw\n' | EDITOR=ed sudo -E systemctl edit --runtime sshd.service
Newline appended
648
654
$ cat /run/systemd/system/sshd.service.d/override.conf 
hello

# New systemctl
$ echo -ne '1a\nhello\n.\nw\n' | EDITOR=ed sudo -E build/systemctl edit --runtime sshd.service
122
128
$ cat /run/systemd/system/sshd.service.d/override.conf 
cat: /run/systemd/system/sshd.service.d/override.conf: No such file or directory
```

Reply:
expecting every user's configured editor to be vi-like is a certified "lol" moment, for sure

Reply:
emacs, nano, vi, pico, joe, … all support the "+" command line syntax for jumping to files too.

"cat" is not an editor. And "ed" is maybe a thought experiment at best.

Not sure we should hold back things just because of niche tools that people just use for the sake of using niche tools...

or to say this differently, maybe if people really want to use "ed" as an edito, they write a script thta eats up the "+" stuff and execs the real thing, and then set $EDITOR to that...

Reply:
So here's the thing: if you hit "v" in "less", then it invokes $VISUAL/$EDITOR with the "+" syntax. If "less" can do that, then so can we.

I guess "less" is quite a standardizing force...

Reply:
"more" does the same btw.

Reply:
"maybe if people want systemd to use the editor they have configured they should change that editor". yeah, maybe. probably not tho.

Reply:
so here's another thing: less doesn't care the least bit about how the editor returns (and you're unlikely to ever even see an error output!), `systemctl edit` errors out if the editor errors out.

Reply:
If all the common editors support the syntax, let's use it.